internal/model/model.go

```diff
@@ -705,19 +705,19 @@ func (m *Model) Close(device protocol.DeviceID, err error) {
 
 // Request returns the specified data segment by reading it from local disk.
 // Implements the protocol.Model interface.
-func (m *Model) Request(deviceID protocol.DeviceID, folder, name string, offset int64, size int, hash []byte, flags uint32, options []protocol.Option) ([]byte, error) {
+func (m *Model) Request(deviceID protocol.DeviceID, folder, name string, offset int64, size int, hash []byte, flags uint32, options []protocol.Option, buf []byte) error {
 	if offset < 0 || size < 0 {
```

Comment:
Passing a buffer with a len different from the `size` parameter, which is a silent weird failure. The two *must* be the same, so either we ought to enforce that, or remove `size` so there's no possibility of a mismatch?